crates/ironrdp-graphics/src/color_conversion.rs

```diff
@@ -15,11 +15,12 @@ pub fn ycbcr_to_bgra(input: YCbCrBuffer<'_>, mut output: &mut [u8]) -> io::Resul
     Ok(())
 }
 
-fn iter_to_ycbcr<'a, I, C>(input: I, y: &mut [i16], cb: &mut [i16], cr: &mut [i16], conv: C)
+fn iter_to_ycbcr<'a, const HINT: usize, I, C>(input: I, y: &mut [i16], cb: &mut [i16], cr: &mut [i16], conv: C)
 where
-    I: IntoIterator<Item = &'a [u8]>,
+    I: ExactSizeIterator<Item = &'a [u8]>,
     C: Fn(&[u8]) -> Rgb,
```

Comment:
question: Can you explain as a in-code comment the purpose of the `HINT` generic parameter? I believe it's a to help diagnostics problems, but is there any rationale for the value it must be set to? Would it make sense to use `debug_assert!` instead?

Reply:
I tried to teach the iterator how many times it can iterate.. assert input.len() == HINT. That didn't help. I should probably remove that hint.